### DISCUSSION TITLE
Settings -> Options says "text" but acts like a "book" note?

### DISCUSSION THREAD

**Discussion by LorenAmelang** [TriliumNext/Trilium#3824] [Q&A]
I was looking for a setting, while using my .cc server on my iPad, and saw this:

![iPad-2023 04 09-01 07 44 000](https://user-images.githubusercontent.com/1297090/230795716-180e7ca0-aee2-4d32-8a2b-5e145be42d09.png)

Pretty obviously says "text", but anything I try to read says "Content of this note cannot be displayed in the book format". 

Same thing happens in the local Windows view. What can this mean? 

It did introduce me to "book notes", which look useful! But when I try converting notes that would make good book notes, it says: 
"It is not recommended to change note type when note content is not empty. Do you want to continue anyway?"

If there is already a note with a nice TOC structure, is there any easy way to make it a book?

**Comment by zadam** [comment]
It's not clear to me what the question is.

Book note basically just acts like an overview of child notes and is useful only for parent notes which don't have any content.

In the case of Options, it's useful only for listing the options titles basically because the content are widgets which are not rendered in this overview.

**Reply by soulsands** [reply]
![image](https://user-images.githubusercontent.com/32272399/230819678-cb910377-753b-47f8-9064-699d49f91f83.png)
He might means the type of options is text, but these option child notes still render in book note context.
Perhaps the options note's type should be fixed to book? @zadam

**Comment by LorenAmelang** [comment]
Yes, I was confused by the Options note behaving differently from typical notes. If I click on one of my ordinary "category" notes, it shows me the child notes contained inside it, in summaries in boxes. I can click one of those and see the actual content. 

(Can't find it now, but seems there is a way to change the format to full-width lines, like my screenshot, instead of boxes? I'd prefer that always!) 

I see now that if I click one of the Options lines anywhere but on the > icon, I get the content, which has note type "widget". Only clicking the > to v icon produces the book error message. 

So is that because Options is acting like a book note? I have a "Searches" category that also acts that way, while the category note is type text:

Search: kyiv#searchString=kyiv ~ancestor=[root](http://127.0.0.1:37840/#root/root) #debug
Content of this note cannot be displayed in the book format

All the boxes there show that "no book" warning. 

--> I guess it depends on the type of the _child_ note - text notes just display, searches or widgets show the "no book" message. 

...

So to my other question, how to make a complex existing note into a book note? 

Maybe: 
Create a new empty note at the category level
Mark it as a book note  (or not - maybe only the child notes need to be "book"?)
Create a new child note under it
Paste in the first TOC item from the existing note
Create another child
Paste another TOC item
...

Seems hardly worth it...  Is there some advantage I haven't discovered? Like maybe export to ePub format? That would be a huge win!

**Reply by zadam** [reply]
> (Can't find it now, but seems there is a way to change the format to full-width lines, like my screenshot, instead of boxes? I'd prefer that always!)

@soulsands has posted a screenshot which shows how to do it. The problem in your case is that "Options" is for some reason a "text" type note which is not the case in normal cases. Maybe you changed it from book to text as part of the experimentation?

Text notes show also "book-like" overview when there's enough space apart from the text itself, but it does not allow configuration like the "proper" book note.

> So to my other question, how to make a complex existing note into a book note?

If by "complex" you mean full of content, then there's no automatic way. You can distribute the content to the children notes manually and then change the parent type to book.

> Seems hardly worth it... Is there some advantage I haven't discovered? Like maybe export to ePub format? That would be a huge win!

There isn't really that much of a benefit honestly. The motivation for this feature was the recognition that people often create "folder" notes, which don't have any content of their own, they act just as a container of other notes. So, instead of these folder notes being blank text notes, they can do something a little bit useful - display an overview of its children notes.

Features like ePub export or similar might happen in the future, but are not really planned ATM.

**Comment by soulsands** [comment]
![Snipaste_2023-04-10_18-16-46](https://user-images.githubusercontent.com/32272399/230882878-30fb23e1-5a8f-4f3b-92ca-e10b6bc06524.png)

@LorenAmelang Maybe you are using an previous version with the book properties tab not showing.
> (Can't find it now, but seems there is a way to change the format to full-width lines, like my screenshot, instead of boxes? I'd prefer that always!)

We can set a note with label #viewType=list to make it render as list like your screenshot, then we can add #expanded to expand all child notes which is what you want.
Book note alway has a viewType label, by default the value is grid. you can change the value to list and toggle collapse/expand.

> So to my other question, how to make a complex existing note into a book note?

![image](https://user-images.githubusercontent.com/32272399/230883503-06a434ba-ccb0-4992-be9e-47e0a3121774.png)
Currently you can click the scissor icon to make it little faster.

**Comment by LorenAmelang** [comment]
Thanks - good info! 
The cc web version on the iPad, and the local Windows version both show 0.59.3, 213, 29. 

I might believe I accidentally changed something on the iPad that first session, because it opened in mobile view, where my choices of Safari font size draw a jumbled overlaid mess. Safari is explicitly configured to use desktop view for Trilium but sometimes it forgets... But before I peeked at Searches I'd forced it to desktop and turned on my mouse - and Searches still showed type = text. 

I just changed the type of both Options and Searches to book, and chose the list option. That synced to the iPad, and both now behave as I'd expect. Noticed Trilium Statistics was already type book - wonder why the others were not? 

`
Text notes show also "book-like" overview when there's enough space apart from the text itself, but it does not allow configuration like the "proper" book note.
`

So they will always show the grid view? 

Just tried setting up a new book note, made the parent note type book, added children with type text, and it does what I'd expect. 


One more detail I was reminded of while testing this...  In Safari showing the cc server, the same view size as my first screenshot above, I can't close a note view by tapping the 'x' at the right of the tab - unless I two-finger zoom in to about 200%. Then it works as expected. I suppose that's some obscure Safari quirk you have no control over, but if you have any way to make the accepted target a bit larger I'd appreciate it. Thankfully if I have the mouse handy it works at the normal view size.

**Reply by zadam** [reply]
> So they will always show the grid view?

Simple answer - Yes. Trilium is to a large degree driven by attributes. If you manually add `#viewType=list` to the text note then the children will display in the list.

If you generally prefer list view, then you can also set root to `#viewType=list` and make it inheritable. Then all these will show as list unless you toggle them explicitly to grid.

**Comment by LorenAmelang** [comment]
This may be unrelated to the book discussion, but I noticed after experimenting with the book note questions that my Windows was really low on memory. Started Process Explorer and saw this:

![Trilium memory](https://user-images.githubusercontent.com/1297090/231026683-0bbb3428-9349-408d-a8cd-829f7f1f697a.PNG)

![image](https://user-images.githubusercontent.com/1297090/231027006-beda737e-e4b6-4042-a00e-608f0cd9c8f3.png)

Restarted Trilium and its memory was back to normal:

![Trilium restarted](https://user-images.githubusercontent.com/1297090/231027126-0de6d853-e5af-402e-b85f-4255e5ed9a2c.PNG)

Any thoughts?

**Reply by zadam** [reply]
Honestly I don't know. The overview displays all its children at once, if you have e.g. large images there, it might take a lot of memory. 

I don't know the exact characteristic of V8 / Chromium, but virtual machines in general have the ugly characteristic that once they allocate memory, they are slow to release it back to the OS. I suspect that's what's going on.

**Reply by LorenAmelang** [reply]
Not sure what "V8 / Chromium" has to do with it - that's the Windows app running directly in Win 10: 
App version:	0.59.3
DB version:	213
Sync version:	29
Build date:	2023-03-27T23:06:14+02:00
Build revision:	[9881e6de3e4966af39ec6245562dca6ac7b25eaa]

My Vivaldi (Chromium) browser definitely absorbs memory and never releases it, as does Thunderbird and Windows itself. I just peeked at Trilium memory use again, app tree closed, only "new tab" showing (but a lot of use last night and "connected standby" since then), and it showed about half that much memory used. Most Trilium items growing by 4K to 10K on every display update, but bringing Trilium to the top for a moment actually dropped memory by about 100K. Grew back to the previous ranges very quickly. 

I guess an 8GB Windows machine  just needs to be restarted after every session. My Linux systems run for months or years without restarting! 

Thanks for listening!

**Reply by zadam** [reply]
> Not sure what "V8 / Chromium" has to do with it - that's the Windows app running directly in Win 10:

The desktop app is built on electron which is essentially a hacked chromium/V8 with node.js.

**Comment by LorenAmelang** [comment]
I just tried this clue, and all my existing top level category notes now show the "list" summaries instead of the "grid" boxes - Yay! 

Show all overviews as list, not grid: Add  #viewType(inheritable)=list  to root→Owned Attributes
https://github.com/zadam/trilium/discussions/3561
